client/app/components/group/_history/history.controller.js

```diff
@@ -49,6 +60,77 @@ class HistoryController {
     return angular.isDefined(this.data.next);
   }
 
+  getStores() {
+    return this.CurrentStores.list.filter((store) => {
+      return angular.isDefined(this.data.results.find((history) => {
+        return history.store === store.id;
+      }));
+    }).map((store) => {
+      if (angular.isUndefined(store._selected)) {
+        store._selected = true;
+      }
+      return store;
+    });
+  }
+
+  getUsers() {
+    return this.CurrentUsers.list.filter((user) => {
+      return angular.isDefined(this.data.results.find((history) => {
+        return history.users.indexOf(user.id) >= 0;
+      }));
+    }).map((user) => {
+      if (angular.isUndefined(user._selected)) {
+        user._selected = true;
+      }
+      return user;
+    });
+  }
+
+  showAllStores(bool) {
+    angular.forEach(this.getStores(), (store) => {
+      store._selected = bool;
```

Comment:
I don't like this, it changes the values in CurrentStores. But everything else seems to involve much more code (e.g. merging together two lists)